src/li.jl

```diff
@@ -0,0 +1,205 @@
+  Hurwitz-zeta function, which is aliased to zeta(s,q) in Julia.
+
+  Accuracy has been tested using many of the identities known for Li
+  and relations to known functions as special cases, and by comparison
+  to `polylog(s, z)` in the `mpmath` arbitrary-precision package in Python. 
+
+      http://mpmath.org/doc/current/functions/zeta.html
+
+  The latter shows deviations of the order of 
+
+  + 10^{Im(s) - 20} for Im(s) < 0
+  + 10^{Im(s) - 20} for Im(s) > 0
+                                                  
+  It isn't clear whether we can do better than this with
+  double-precision arithmetic.
+
+## Arguments
+* `s::Complex`: the 'fractional' coefficient
+* `z::Complex`: the point at which to calculate it
+* `accuracy::Real=1.0e-18`: nominal accuracy of calculation, but mainly useful for testing
+
+## Examples
+```jldoctest
+julia> Li(-1.0, 0.0) 
+(0.0,1)
+```
+"""
+function Li(s::Number, z::Number, accuracy::Real=1.0e-18)
+    T = 0.5 # threshold at which we change algorithms
+    if z ≈ 1.0
+        if real(s) > 1
+            return zeta(s)
+        else
+            return Inf
+        end
+    elseif z ≈ -1.0
+        return -eta(s)
+    elseif s ≈ 0.0
+        return z ./ (1-z)
+    elseif abs(z) <= T
+        return Li_direct(s, z, accuracy)
+    elseif abs(z) >= 1/T && isinteger(s) && real(s) < 0
+        # use inversion formula to calculate in terms of Li_n(1/z)
+        # but note for negative integer s, it collapses to something small
+        return -(-1.0)^s .*Li_direct(s, 1/z, accuracy)
+    elseif  abs(z) >= 1/T
+        # use inversion formula to calculate in terms of Li_s(1/z)
+        G = (2*pi*im).^s .* zeta( 1-s, 0.5 - log(-1/complex(z))/(2*pi*im) ) ./  gamma(s)
+        F = complex(-1.0)^s .* Li_direct(s, 1/z, accuracy)
+        A = 2*pi*im*log(complex(z))^(s-1)/(gamma(s))
+        if ( isreal(z) && real(z)>=1 )
+            Θ = 1
+        else
+            Θ = 0
+        end
+        # println("G = $G, F=$F, Θ=$Θ, A=$A")
+        return ( G - F - Θ*A )
+    else 
+        # power series around mu=0, for z = e^mu
+        Li_series_mu(s, z, accuracy)
+    end
+end
+     
+
+"""
+    Polylog(s, z)
+
+    This is just an alias of Li(s, z)
+
+"""
+polylog(s::Number, z::Number ) = Li(s, z)
+
+    
+####################################################################
+#### these are component functions and aren't exported at this point
+     
+# Dirichlet beta function, for testing results
+#   https://en.wikipedia.org/wiki/Dirichlet_beta_function
+#   but don't export as it would create confusion with Euler integral     
+function Dbeta(s::Number)
+    β = 4.0^(-s) * ( zeta(s,0.25) - zeta(s,0.75) )
+end
+    
+function Li_direct(s::Number, z::Number, accuracy=1.0e-18)
+    # calculate using direct definition
+    if abs(z) > 1 || ( abs(z) ≈ 1  && real(s) <= 2)
+        warn("Should have |z| < 1 or (|z|==1 and Re(s)>2)")
+        throw(DomainError)
+    end
```

Comment:
In this case, since this is a restriction on the arguments to the function, you can (and should) use `ArgumentError`. That is,
```julia
if abs(z) > 1 || (abs(z) ≈ 1 && real(s) <= 2)
    throw(ArgumentError("|z| < 1 or (|z| == 1 and Re(s) > 2) not satisfied"))
end
```
Most typed exception constructors accept strings which can provide specific messages. However, for whatever reason, `DomainError` does not.